lib/application.js

```diff
@@ -567,3 +571,65 @@ app.listen = function(cb) {
 
   return server;
 };
+
+/* START CONTROLLER CODE*/
+
+// this is similar to app.model
+// at the moment I haven't really used 'options'
+app.controller = function(ctor, options) {
+  var self = this;
+
+  // the SharedController is a wrapper, see below
+  var controller = new SharedController(ctor);
+  this.remotes().addClass(controller.sharedClass);
+  app.controllers().push(controller);
+
+  return controller;
+
+  // SharedController class contains strong-remoting objects to set up the metadata
+  // and register the endpoint, etc
+  function SharedController(ctor, options) {
+    options = options || {};
+    this.name = ctor.name;
+    this.ctor = ctor;
+    // TODO - right now there's a hack on line 289 to skip this
+    // util.inherits(ctor, EventEmitter);
+
+    this.sharedClass = new RemoteObjects.SharedClass(this.name, ctor, options);
```

Comment:
Taking into account https://github.com/strongloop/loopback/pull/2284/files#r62626126, this may be a good solution:

```js
this.name = ctor.controllerName = options.name || ctor.controllerName || ctor.name;
```